packages/zosfiles/__tests__/__system__/methods/upload/Upload.system.test.ts

```diff
@@ -1023,6 +1024,35 @@ describe("Upload USS file", () => {
             // Compare file view with matching upload and view encoding (1047).
             expect(readResponseGood.stdout.toString()).toContain(fileContents);
         });
+
+        it("should upload a USS file with special characters and verify correct chunked upload/download", async () => {
+            const testFilePath = path.join(__dirname, "testfiles", "specialCharTestData.txt");
+            const targetUssPath = ussname;
+            const file = path.basename(targetUssPath);
+            const localDirname = `${testEnvironment.workingDir}/ussDir`;
+            const downloadedFilePath = path.join(localDirname, file);
+
+            const originalContent = fs.readFileSync(testFilePath, "utf-8");
+            const expectedUniqueChars = new Set(stripNewLines(originalContent));
+
+            await Upload.fileToUssFile(
+                REAL_SESSION,
+                testFilePath,
+                targetUssPath,
+                { encoding: "UTF-8" }
+            );
+
+            await Download.ussFile(
+                REAL_SESSION,
+                targetUssPath,
+                { encoding: "UTF-8", file: downloadedFilePath }
+            );
+
+            const downloadedContent = fs.readFileSync(downloadedFilePath, "utf-8");
+            const downloadedUniqueChars = new Set(stripNewLines(downloadedContent));
+
+            expect(downloadedUniqueChars).toStrictEqual(expectedUniqueChars);
```

Comment:
(nit) we could just check if the contents are equal after stripping new lines, as there is logical overhead in the Set that causes each duplicate element to be stripped. Doing a `toStrictEqual` check between the two contents (before de-duplication) should have a similar effect w/ a negligible performance difference.